File: docs/src/testers/functions/management/get_function_details.ts
// @sniptest filename=get_function_details.ts
// @sniptest show=4-17
import assert from 'node:assert/strict';
const functionId = process.env.NOTTE_FUNCTION_ID!;

import { NotteClient } from 'notte-sdk';

const client = new NotteClient();

// Get function by ID
const fn = client.NotteFunction({ function_id: functionId });
const response = await fn.get();

// Access function properties
console.log(`Function ID: ${response.function_id}`);
console.log(`Name: ${response.name}`);
console.log(`Description: ${response.description}`);
console.log(`Latest Version: ${response.latest_version}`);
console.log(`Versions: ${response.versions}`);

assert.equal(response.function_id, functionId);
assert.ok(response.versions.includes(response.latest_version));
const results = [response.function_id === functionId, response.versions.length > 0];
export { results };

File: docs/src/testers/functions/management/get_function_details.py
# @sniptest filename=get_function_details.py
# @sniptest show=5-17
# ruff: noqa: E402 -- live fixture precedes displayed import
import os

function_id = os.environ["NOTTE_FUNCTION_ID"]
from notte_sdk import NotteClient

client = NotteClient()

# Get function by ID
function = client.Function(function_id=function_id)

# Access function properties
print(f"Function ID: {function.function_id}")
print(f"Name: {function.response.name}")
print(f"Description: {function.response.description}")
print(f"Latest Version: {function.response.latest_version}")
print(f"Versions: {function.response.versions}")

assert function.function_id == function_id
assert function.response.latest_version in function.response.versions
results = [function.function_id == function_id, bool(function.response.versions)]
